astropy/time/core.py

```diff
@@ -1269,13 +1310,13 @@ def __sub__(self, other):
         # we need a constant scale to calculate, which is guaranteed for
         # TimeDelta, but not for Time (which can be UTC)
         if other_is_delta:  # T - Tdelta
+            out = self.replicate()
             if self.scale in other.SCALES:
-                out = self.replicate()
                 if other.scale not in (out.scale, None):
                     other = getattr(other, out.scale)
             else:
-                out = getattr(self, (other.scale if other.scale is not None
-                                     else 'tai'))
+                out = getattr(out, (other.scale if other.scale is not None
```

Comment:
Yes, I can see that this took some time to track down! Maybe in this case the code would be clearer if instead of using `getattr` we just did `out._set_scale(...)`?

Quick PR to your branch forthcoming...